audit/hashstructure_test.go

```diff
@@ -98,20 +98,28 @@ func TestCopy_response(t *testing.T) {
 	}
 }
 
-func TestHashString(t *testing.T) {
+type TestSalter struct{}
```

Comment:
go doc comments: 👍 
question ❓ : I don't think that is necessary, since we are likely to use the **TestSalter** type somewhere in the same set of tests that a **Salter** interface is expected. If **TestSalter** doesn't satisfy the interface, then it will generate a compiler error. Using the `var _ Salter = (*TestSalter)(nil)` expression would be done if the TestSalter struct was defined in a module but not used as a **Salter** interface in that module. In that case, there wouldn't be any compiler errors when building the module where **TestSalter** is defined. However, when some other code imports the **TestSalter**'s module and tries to use a **TestSalter** instance where a **Salter** interface is required, that would generate a compiler error.